include/oneapi/dpl/pstl/numeric_impl.h

```diff
@@ -206,13 +206,19 @@ __brick_transform_scan(_RandomAccessIterator __first, _RandomAccessIterator __la
                        /*is_vector=*/::std::true_type) noexcept
 {
 #if (_PSTL_UDS_PRESENT || _ONEDPL_UDS_PRESENT)
-    return __unseq_backend::__simd_scan(__first, __last - __first, __result, __unary_op, __init, __binary_op,
-                                        _Inclusive());
-#else
+    const auto& __first_tmp = *__first;
+    const auto& __result_tmp = *__result;
+    // An in-place scan violates the intra-iteration dependency restriction of the omp scan directive. We must call our serial brick.
+    if (static_cast<const void*>(::std::addressof(__first_tmp)) !=
```

Comment:
Unfortunately,  a comparison of pointers is not reliable here, due to a dereferenced result might be a kind of proxy objects... For example, in case of output transform iterator or something like that,